# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/canvas/WebGLContext.h
# Commit: 71ad2ed8e9ba
# Full Hash: 71ad2ed8e9ba668e11656652ebd4235ef64565c2
# Author: Jeff Gilbert <jgilbert@mozilla.com>
# Date: 2020-06-11 04:04:33
# Regressor Bug: 1632249
# File Overlap Count: 1
# Description:
#   Bug 1632249 - Create a separate WebGLContext::PresentInto function for WebXR. r=jgilbert
#   
#   - Do not apply mIsCanvasDirty bailout when presenting from a opaque FBO
#   back buffer.
#   - Create a separate WebGLContext::PresentInto with more specific XR
# ==============================================================================

diff -r 6e9a89ead3a5 -r 71ad2ed8e9ba dom/canvas/WebGLContext.h
--- a/dom/canvas/WebGLContext.h	Wed Jun 10 22:21:04 2020 +0000
+++ b/dom/canvas/WebGLContext.h	Wed Jun 10 22:21:07 2020 +0000
@@ -101,7 +101,7 @@
 namespace layers {
 class CompositableHost;
 class SurfaceDescriptor;
-}
+}  // namespace layers
 
 namespace webgl {
 class AvailabilityRunnable;
@@ -483,6 +483,7 @@
   // Present to compositor
  private:
   bool PresentInto(gl::SwapChain& swapChain);
+  bool PresentIntoXR(gl::SwapChain& swapChain, const gl::MozFramebuffer& fb);
 
  public:
   void Present(WebGLFramebuffer*, layers::TextureType);
